src/services/redisService.js

```diff
@@ -0,0 +1,495 @@
+                    }
+                }
+
+                if (!this._shouldReconnectOnError(error) || attempt === maxRetries) {
+                    console.error(`Max retries reached for storing file ${fileKey} on pod: ${this.podId}`);
+                    return false;
+                }
+
+                if (attempt < maxRetries) {
+                    const delay = Math.min(1000 * Math.pow(2, attempt - 1), 5000); // Max 5s delay
+                    console.log(`⏳ Waiting ${delay}ms before retry ${attempt + 1}...`);
+                    await new Promise(resolve => setTimeout(resolve, delay));
+                }
+            }
+        }
+    }
+
+    async storeFile(fileKey, filePath, ttl) {
+        if (!this.isConnected || !this.redis) {
+            throw new Error(`Redis not connected on pod: ${this.podId}`);
+        }
+
+        const fileExists = await fs.promises.access(filePath).then(() => true).catch(() => false);
+        if (!fileExists) {
+            throw new Error(`File does not exist: ${filePath}`);
+        }
+
+        const stats = await fs.promises.stat(filePath);
+        if (stats.size > this.maxFileSize) {
+            throw new Error(`File too large: ${stats.size} bytes (max: ${this.maxFileSize})`);
+        }
+
+        console.debug(`Reading file for Redis storage: ${filePath} (${stats.size} bytes)`);
+        const fileBuffer = await fs.promises.readFile(filePath);
+
+        const metadata = {
+            originalName: path.basename(filePath),
+            size: stats.size,
+            mimeType: this._getMimeType(filePath),
+            uploadedAt: new Date().toISOString(),
+            uploadedBy: this.podId,
+            ttl: ttl
+        };
+
+        console.log(`Storing ${fileBuffer.length} bytes in Redis with key: ${fileKey}`);
+    
+        const pipeline = this.redis.pipeline();
+        pipeline.setex(`${fileKey}:data`, ttl, fileBuffer);
+        pipeline.setex(`${fileKey}:metadata`, ttl, JSON.stringify(metadata));
+
+        const results = await Promise.race([
+            pipeline.exec(),
+            new Promise((_, reject) => 
+                setTimeout(() => reject(new Error('Redis pipeline timeout')), 30000) // 30s timeout
+            )
+        ]);
+
+        const dataStored = results[0][1] === 'OK';
+        const metadataStored = results[1][1] === 'OK';
+
+        if (!dataStored || !metadataStored) {
+            throw new Error(`Redis pipeline failed - data: ${dataStored}, metadata: ${metadataStored}`);
+        }
+
+        console.log(`File stored in Redis successfully: ${fileKey} (${stats.size} bytes) on pod: ${this.podId}`);
+
+        const verified = await this.verifyStoredFile(fileKey, stats.size);
+        if (!verified) {
+            throw new Error('File storage verification failed');
+        }
+
+        return true;
+    }
+
+    async verifyStoredFile(fileKey, expectedSize, maxRetries = 3) {
```

Comment:
Hence, this is again going to attempt to connect to Redis; better to verify the connection